e2e/playwright/viewer/src/tests/viewer-action.spec.ts

```diff
@@ -0,0 +1,163 @@
+  });
+
+  test('[C297585] Cancel Editing action', async ({ personalFiles }) => {
+    await personalFiles.dataTable.performClickFolderOrFileToOpen(fileForCancelEditing);
+    await personalFiles.viewer.waitForViewerToOpen();
+    await personalFiles.acaHeader.clickViewerMoreActions();
+    await personalFiles.matMenu.clickMenuItem('Cancel Editing');
+    await personalFiles.reload({ waitUntil: 'domcontentloaded' });
+    await personalFiles.acaHeader.clickViewerMoreActions();
+    expect(await personalFiles.matMenu.isMenuItemVisible('Edit Offline'), 'Edit offline menu should be visible').toBe(true);
+  });
+
+  test('[C279282] Full screen action', async ({ personalFiles }) => {
+    await personalFiles.dataTable.performClickFolderOrFileToOpen(randomDocxName);
+    await personalFiles.viewer.waitForViewerToOpen();
+    await personalFiles.acaHeader.fullScreenButton.click();
+    expect(await personalFiles.viewer.isViewerOpened(), 'Viewer is closed after pressing Full screen').toBe(true);
+  });
+
+  test('[C286314] Pressing ESC in the viewer closes only the action dialog', async ({ personalFiles }) => {
+    await personalFiles.dataTable.performClickFolderOrFileToOpen(randomDocxName);
+    await personalFiles.viewer.waitForViewerToOpen();
+    await personalFiles.acaHeader.clickViewerMoreActions();
+    await personalFiles.viewerDialog.clickActionsCopy();
+    expect(await personalFiles.viewerDialog.isCopyDialogOpen(), 'Dialog is not open').toBe(true);
+    await personalFiles.page.keyboard.press('Escape');
+    expect(await personalFiles.viewerDialog.isCopyDialogClose(), 'Dialog is not open').toBe(false);
+    expect(await personalFiles.viewer.isViewerOpened(), 'Viewer should be opened').toBe(true);
+  });
+
+  test('[C286379] Favorite action from Shared Files', async ({ sharedPage, favoritePage, shareAction }) => {
+    await sharedPage.navigate({ waitUntil: 'domcontentloaded' });
+    await sharedPage.dataTable.performClickFolderOrFileToOpen(randomDocxNameShare);
+    expect(await sharedPage.viewer.isViewerOpened(), 'Viewer should be opened').toBe(true);
+
+    await sharedPage.acaHeader.clickViewerMoreActions();
+
+    await favoritePage.viewerDialog.favoriteMenuButton.waitFor({ state: 'attached', timeout: timeouts.normal });
+    await sharedPage.viewerDialog.favoriteMenuButton.click();
+    await favoritePage.viewerDialog.favoriteMenuButton.waitFor({ state: 'detached', timeout: timeouts.normal });
+
+    await sharedPage.acaHeader.clickViewerMoreActions();
+    expect(await sharedPage.viewerDialog.removeFavoriteMenuButton.isVisible(), 'Item should be remove favorite').toBe(true);
+    await sharedPage.page.keyboard.press('Escape');
+    await favoritePage.navigate({ waitUntil: 'domcontentloaded' });
+    expect(await favoritePage.dataTable.isItemPresent(randomDocxNameShare), 'Item is not present in Favorites list').toBe(true);
+    expect(await shareAction.isFavorite(randomDocxNameShare, 'hruser'), 'Item is not favorite').toBe(true);
```

Comment:
Should we do API validation in UI test?

Reply:
Here, Its been done in protractor test, but we can remove this extra validation as not needed.